siliconcompiler/tools/yosys/syn_fpga.tcl

```diff
@@ -4,19 +4,159 @@ set job_name [dict get $sc_cfg option jobname]
 set step [dict get $sc_cfg arg step]
 set index [dict get $sc_cfg arg index]
 
+set sc_memmap_techlib "None"
+set sc_mem_techlib "None"
+set sc_lut_techlib "None"
+set sc_dsp_techlib "None"
+set sc_flop_techlib "None"
+set sc_syn_lut_size [dict get $sc_cfg tool $sc_tool task $sc_step var lut_size ]
+
 #TODO: add logic that remaps yosys built in name based on part number
 
 # Run this first to handle module instantiations in generate blocks -- see
 # comment in syn_asic.tcl for longer explanation.
 yosys hierarchy -top $sc_design
 
+#***NOTE:  There are over a dozen customized synthesis routines in yosys
+#          for different FPGA families.  The level of maturity and the
+#          range of parts supported by each varies significantly:  some
+#          are experimental, some support only one part, others are mature,
+#          and the Xilinx and Intel setups specifically have -family flags
+#          to select different parts within those companies' product
+#          portfolios.
+#          As of 1/13/2023, what's here is a best-effort first stab at getting
+#          all of the commands exercised in a first-order fashion and defining
+#          part number strings that could be used to specify each supported
+#          part number.  Where available, part number prefixes have been gleaned
+#          from DigiKey so that there is a correspondence here between
+#          real product part numbers and the if/else statement selecting a
+#          synthesis command.  It will be necessary to do some vetting and
+#          decide what to really support and what to omit based on the yosys
+#          support (or the place and route support) being too skimpy.
+#          -PG 1/13/2023
```

Comment:
We should be consistent in our code base. afaik there are no other examples of file authors or blocks/signatures, so these should be fixed.  (ie no need for PG 1/13/2023)

Reply:
Done